src/Weather/Skysight/Request.hpp

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright_License {
+
+  XCSoar Glide Computer - http://www.xcsoar.org/
+  Copyright (C) 2000-2016 The XCSoar Project
+  A detailed list of copyright holders can be found in the file "AUTHORS".
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+#ifndef WEATHER_SKYSIGHTREQUEST_HPP
+#define WEATHER_SKYSIGHTREQUEST_HPP
+
+#include "APIGlue.hpp"
+#include "Thread/StandbyThread.hpp"
+#include "Util/tstring.hpp"
+
+#include "OS/Path.hpp"
+#include "OS/FileUtil.hpp"
+
+#include "Net/HTTP/Session.hpp"
+#include "Net/HTTP/Request.hpp"
+#include "Net/HTTP/Handler.hpp"
+
+#include "IO/FileLineReader.hpp"
+
+#include "LogFile.hpp"
+
+
+class SkysightRequestError {};
```

Comment:
Needs documentation. Since this is not derived from `std::exception`, the usual `catch` blocks won't catch it.